### PR TITLE
JP-1802 Add new dither keyword subpxpts to constraints

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [JP-1802](https://jira.stsci.edu/browse/JP-1802) / #5523

### CODE REVIEW

File: jwst/associations/lib/rules_level2b.py [hunk 1]
```diff
@@ -760,8 +760,8 @@ def __init__(self, *args, **kwargs):
                                 sources=['numdthpt'],
                             ),
                             DMSAttrConstraint(
-                                name='subpxpns',
-                                sources=['subpxpns'],
+                                name='subpxpts',
+                                sources=['subpxpns', 'subpxpts'],
```
Comment (hbushouse) [inline jwst/associations/lib/rules_level2b.py:764]: Yay for backwards compatibility!

**Review by hbushouse — APPROVED** [review]
Requisite reminder for change log entry ...

File: jwst/associations/lib/rules_level2b.py (hunk 1 above)
Comment (stscieisenhamer) [reply jwst/associations/lib/rules_level2b.py:764]: and, believe it or not, by design!

Comment (hbushouse) [reply jwst/associations/lib/rules_level2b.py:764]: And what a clever design. One instance here that lists both keywords as sources can then be referenced anywhere else simply using the single constraint name "subpxpts" and it works like magic!

**Review by tapastro — APPROVED** [review]
☑️ 

This got me reading associations.lib.constraint, which I might have to save for another day...

**Pull request merged**